Cesium3DTilesContent/src/SubtreeAvailability.cpp

```diff
@@ -1,465 +1,194 @@
+          subtree.childSubtreeAvailability,
+          subtree.buffers,
+          subtree.bufferViews);
+  if (!maybeChildSubtreeAvailability)
     return std::nullopt;
-  }
 
-  std::vector<SubtreeBufferView> bufferViews;
-  auto bufferViewIt = subtreeJson.FindMember("bufferViews");
-  if (bufferViewIt != subtreeJson.MemberEnd() &&
-      bufferViewIt->value.IsArray()) {
-    bufferViews.resize(bufferViewIt->value.Size());
-    for (rapidjson::SizeType i = 0; i < bufferViewIt->value.Size(); ++i) {
-      const auto& bufferViewJson = bufferViewIt->value[i];
-      auto bufferIdxIt = bufferViewJson.FindMember("buffer");
-      auto byteOffsetIt = bufferViewJson.FindMember("byteOffset");
-      auto byteLengthIt = bufferViewJson.FindMember("byteLength");
-
-      if (bufferIdxIt == bufferViewJson.MemberEnd() ||
-          !bufferIdxIt->value.IsUint()) {
-        return std::nullopt;
-      }
-
-      if (byteOffsetIt == bufferViewJson.MemberEnd() ||
-          !byteOffsetIt->value.IsUint()) {
-        return std::nullopt;
-      }
-
-      if (byteLengthIt == bufferViewJson.MemberEnd() ||
-          !byteLengthIt->value.IsUint()) {
-        return std::nullopt;
-      }
-
-      bufferViews[i].bufferIdx = bufferIdxIt->value.GetUint();
-      bufferViews[i].byteOffset = byteOffsetIt->value.GetUint();
-      bufferViews[i].byteLength = byteLengthIt->value.GetUint();
-    }
-  }
-
-  auto tileAvailability =
-      parseAvailabilityView(tileAvailabilityIt->value, buffers, bufferViews);
-  if (!tileAvailability) {
+  // At least one element is required in contentAvailability.
+  if (subtree.contentAvailability.empty())
     return std::nullopt;
-  }
-
-  auto childSubtreeAvailability = parseAvailabilityView(
-      childSubtreeAvailabilityIt->value,
-      buffers,
-      bufferViews);
-  if (!childSubtreeAvailability) {
-    return std::nullopt;
-  }
 
-  std::vector<AvailabilityView> contentAvailability;
-  if (contentAvailabilityIt->value.IsArray()) {
-    contentAvailability.reserve(contentAvailabilityIt->value.Size());
-    for (const auto& contentAvailabilityJson :
-         contentAvailabilityIt->value.GetArray()) {
-      auto availability =
-          parseAvailabilityView(contentAvailabilityJson, buffers, bufferViews);
-      if (!availability) {
-        return std::nullopt;
-      }
+  std::vector<SubtreeAvailability::AvailabilityView> contentAvailability;
+  contentAvailability.reserve(subtree.contentAvailability.size());
 
-      contentAvailability.emplace_back(*availability);
-    }
-  } else {
-    auto availability = parseAvailabilityView(
-        contentAvailabilityIt->value,
-        buffers,
-        bufferViews);
-    if (!availability) {
-      return std::nullopt;
+  for (const auto& availabilityDesc : subtree.contentAvailability) {
+    auto maybeAvailability = parseAvailabilityView(
+        availabilityDesc,
+        subtree.buffers,
+        subtree.bufferViews);
+    if (maybeAvailability) {
```

Comment:
That would be a good change, but I don't think it needs to happen in this PR because it wasn't done in the original code, either. In general, cesium-native is a loader, not a 3D Tiles validator, so the only _promise_ we make when loading malformed data is that we won't crash.

Reply:
Verified safety checks when accessing `_contentAvailability`